### ISSUE TITLE
Invalid inputs should be accessible

### ISSUE THREAD

**Issue by juanca** [mavenlink/design-system#67]
I've been putting off further work into invalid input states in favor of using a CSS class -- all in the while collecting information about how to best implement and test invalidity. 

I'm keeping this issue to track that work. 

- [HTML5 Constraint validation API](https://developer.mozilla.org/en-US/docs/Web/API/Constraint_validation) can set custom invalid states without subclassing HTMLInputElement
- [`@testing-library/react` matcher](https://github.com/testing-library/jest-dom#tobeinvalid) can check against `aria-invalid` or `checkValidity()`

I think it would be best to deal with this sooner than later since we're about to have ~5 `<input>` and ~2 `<select>` components.
